### PR TITLE
feat(uptime): Implement delete endpoint for uptime monitors

### PR DESCRIPTION
**Pull request by wedamija**
This implements an api for deleting uptime monitors. Also renames `delete_project_uptime_subscription` since it wasn't well named

### CODE REVIEW

File: src/sentry/uptime/endpoints/project_uptime_alert_details.py [hunk 1]
```diff
@@ -95,3 +97,27 @@ def put(
             return self.respond(validator.errors, status=400)
 
         return self.respond(serialize(validator.save(), request.user))
+
+    @extend_schema(
+        operation_id="Update an Uptime Monitor for a Project",
```
Comment (evanpurkhiser) [inline src/sentry/uptime/endpoints/project_uptime_alert_details.py:102]: Description wrong?

Comment (evanpurkhiser) [reply src/sentry/uptime/endpoints/project_uptime_alert_details.py:102]: Also why is it "id" if it's human text?

File: src/sentry/uptime/endpoints/project_uptime_alert_details.py [hunk 2]
```diff
@@ -95,3 +97,27 @@ def put(
...
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            GlobalParams.PROJECT_ID_OR_SLUG,
+            UptimeParams.UPTIME_ALERT_ID,
+        ],
+        request=UptimeMonitorValidator,
```
Comment (evanpurkhiser) [inline src/sentry/uptime/endpoints/project_uptime_alert_details.py:108]: Is this right?

File: src/sentry/uptime/subscriptions/subscriptions.py [hunk 3]
```diff
@@ -101,6 +101,12 @@ def delete_project_uptime_subscription(
     remove_uptime_subscription_if_unused(uptime_subscription)
 
 
+def delete_project_uptime_subscription(subscription: ProjectUptimeSubscription):
```
Comment (evanpurkhiser) [inline src/sentry/uptime/subscriptions/subscriptions.py:104]: Should we record a audit entry?

File: src/sentry/uptime/endpoints/project_uptime_alert_details.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/uptime/endpoints/project_uptime_alert_details.py:102]: Idk, but all the apis use it this way

File: src/sentry/uptime/endpoints/project_uptime_alert_details.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/uptime/endpoints/project_uptime_alert_details.py:108]: Too lazy with my copy/pasting

File: src/sentry/uptime/subscriptions/subscriptions.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/uptime/subscriptions/subscriptions.py:104]: Oh yeah, I forgot about this here!

**Review by evanpurkhiser — APPROVED** [review]
few minor things. Looks good

**Pull request merged**